### PR TITLE
Add support for sending signals in stress tests

### PR DESCRIPTION
**Pull request by Harivallabha-Rangarajan**
- Add a sendSignals() function to LoadTestDataStore, to add support for sending signals in stress tests
- Refactor support for sending ops in stress tests into a sendOps() function
- Add a profile for testing scale with signals in testConfig.json

### CODE REVIEW

**Comment by Harivallabha-Rangarajan** [comment]
[Label external PRs](https://github.com/microsoft/FluidFramework/runs/7442801141?check_suite_focus=true)
GitHub Actions has encountered an internal error when running your job.

Any idea how I could resolve this? :")

**Comment by anthony-murphy** [comment]
@tyler-cai-microsoft added a signal workload in stress already. see LeaderElection. i'm not sure another is necessary, if there is something new here can we unify the two

**Comment by Harivallabha-Rangarajan** [comment]
@anthony-murphy 

Just took a look at that, thanks! But I think that the two workloads serve very different purposes. Having sendOps(), and sendSignals() methods as implemented above helps us run tests for whiteboard simulations where we want to stress test something like 10k users concurrently sending 0.6 signals/sec and 0.1 ops/sec.

**Comment by Harivallabha-Rangarajan** [comment]
Rebasing on top of latest changes from upstream.

**Comment by Harivallabha-Rangarajan** [comment]
@anthony-murphy I've rebased the changes on top of the latest upstream commits. Could you please take a look when you get a chance? Thanksss!

File: packages/test/test-service-load/src/runner.ts
Comment (anthony-murphy) [inline packages/test/test-service-load/src/runner.ts:403]: why don't we want to count the receiving of our own singals?

File: packages/test/test-service-load/src/runner.ts
Comment (anthony-murphy) [inline packages/test/test-service-load/src/runner.ts:395]: we should combine the event handlers here. not need to have two different ones on the same event name

File: packages/test/test-service-load/testConfig.json
Comment (anthony-murphy) [inline packages/test/test-service-load/testConfig.json:121]: can you add this to the ci profile above as well, so that we run it too. i think just matching the ops rate should be fine

File: packages/test/test-service-load/src/loadTestDataStore.ts
Comment (anthony-murphy) [inline packages/test/test-service-load/src/loadTestDataStore.ts:444]: in the future try to avoid mixing refactoring with changes, it makes it harder to review as i need to parse refactors from changes. if you really want  a refactor just push a small pr with only that. generally quick and easy to review as long as it is non-controversial like this

File: packages/test/test-service-load/src/loadTestDataStore.ts
Comment (anthony-murphy) [inline packages/test/test-service-load/src/loadTestDataStore.ts:513]: we primarily do this in case there is throttling, so the client has a rest period to recover. do we not throttle on signals? @GaryWilber  do you know?

File: packages/test/test-service-load/src/runner.ts
Comment (Harivallabha-Rangarajan) [reply packages/test/test-service-load/src/runner.ts:403]: Local signals are handled locally, and are not sent to PUSH so we don't count them.

File: packages/test/test-service-load/src/loadTestDataStore.ts
Comment (Harivallabha-Rangarajan) [reply packages/test/test-service-load/src/loadTestDataStore.ts:444]: Sounds good, I'll keep that in mind :")

File: packages/test/test-service-load/src/loadTestDataStore.ts
Comment (Harivallabha-Rangarajan) [reply packages/test/test-service-load/src/loadTestDataStore.ts:513]: We didn't notice such throttling on signals during our test runs - so I think it may not be necessary.

File: packages/test/test-service-load/src/loadTestDataStore.ts
Comment (vikagrawal) [inline packages/test/test-service-load/src/loadTestDataStore.ts:455]: Nit ' runResult is of type ....'

File: packages/test/test-service-load/src/loadTestDataStore.ts
Comment (vikagrawal) [inline packages/test/test-service-load/src/loadTestDataStore.ts:496]: Passing the timeout to two functions and clearing it in both. Better will be clear it in same function from where it's created and passed. Additionally, if there is an exception then timeout will not be cleared.

Comment (Harivallabha-Rangarajan) [reply packages/test/test-service-load/src/loadTestDataStore.ts:496]: Done!

File: packages/test/test-service-load/src/loadTestDataStore.ts
Comment (Harivallabha-Rangarajan) [reply packages/test/test-service-load/src/loadTestDataStore.ts:455]: Gotcha, thanks!

File: packages/test/test-service-load/testConfig.json
Comment (Harivallabha-Rangarajan) [reply packages/test/test-service-load/testConfig.json:121]: Done!

File: packages/test/test-service-load/src/runner.ts
Comment (Harivallabha-Rangarajan) [reply packages/test/test-service-load/src/runner.ts:395]: Done!

**Review by vikagrawal — APPROVED** [review]

**Comment by taylorsw04** [comment]
@anthony-murphy should this change be merged? I saw a comment that might be waiting on @GaryWilber to reply to, but otherwise it seems complete.

File: packages/test/test-service-load/src/loadTestDataStore.ts
Comment (GaryWilber) [reply packages/test/test-service-load/src/loadTestDataStore.ts:517]: There is also throttling for signals and the throttle limit is higher since signals are cheaper for the system

**Comment by anthony-murphy** [comment]
> @anthony-murphy should this change be merged? I saw a comment that might be waiting on @GaryWilber to reply to, but otherwise it seems complete.

looks good. i was unexpected away for a bit, so lost track of this.

**Comment by Harivallabha-Rangarajan** [comment]
Yes: I don't have merge access, so I need someone to merge it. @vikagrawal can you please help me merge this PR? Thankss! :")

**Pull request merged**
